src/jackdaw/test/fixtures.clj

```diff
@@ -162,6 +164,71 @@
                                 timeout)
                         {}))))))
 
+(defn delete-recursively [fname]
+  (let [func (fn [func f]
+               (when (.isDirectory f)
+                 (doseq [f2 (.listFiles f)]
+                   (func func f2)))
+               (clojure.java.io/delete-file f))]
+    (func func (clojure.java.io/file fname))))
+
+(defn empty-state-fixture [app-config]
+  (fn [t]
+    (let [state-dir (format "%s/%s"
+                            (or (get app-config "state.dir")
+                                "/tmp/kafka-streams")
+                            (get app-config "application.id"))]
+      (when (.exists (io/file state-dir))
+        (log/info "deleting state dir: " state-dir)
+        (delete-recursively state-dir))
+      (t))))
+
+;;; reset-application-fixture ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn reset-application-fixture [app-config]
```

Comment:
In the reloaded workflow, Stuart Sierra defines "reset" this way:
> Finally, the heart of my workflow: the reset function. This is one function which I can call at the REPL to 1) stop the current application instance; 2) reload any source files that have changed; and 3) create and start a new application instance.

In `user/reset`, we don't reload source files, but we do stop the app, clean up, create topics, and restart. This seems closer to the `integration-fixture` fixture than the `reset-application-fixture`. How should we align the two approaches?

Reply:
The "reset" terminology here has nothing to do with re-loaded. It's the [application reset tool](https://docs.confluent.io/current/streams/developer-guide/app-reset-tool.html) from kafka

Reply:
Let's hold this until the next PR as discussed.